python/tvm/testing.py

```diff
@@ -285,4 +288,105 @@ def _check_forward(constraints1, constraints2, varmap, backvarmap):
                    constraints_trans.dst_to_src, constraints_trans.src_to_dst)
 
 
+def gpu(f):
+    """Mark to differentiate tests that use the GPU is some capacity. These
+    tests will be run on CPU-only nodes and on nodes with GPUS.
+
+    To mark a test that must have a GPU present to run, use `@requires_gpu`.
+    """
+    return pytest.mark.gpu(f)
+
+
+def requires_gpu(f):
```

Comment:
Perhaps it is more useful to mark a region besides mark a function. e.g. 

```python
with tvm.testing.cuda_region():
    pass
```

Reply:
I don't think this is possible with pytest. We don't want to have to run every test to see if they require a gpu or not.